File: Java/88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int first = m - 1, second = n - 1, entire = m + n - 1;
        while (first >= 0 && second >= 0) {
            nums1[entire--] = nums1[first] > nums2[second] ? nums1[first--] : nums2[second--];
        }
        while (second >= 0) {
            nums1[entire--] = nums2[second--];
        }
    }
}

File: Python/88.py
from typing import List


class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        last_index = n + m - 1
        i, j = m - 1, n - 1

        while j >= 0:
            if i >= 0 and nums1[i] > nums2[j]:
                nums1[last_index] = nums1[i]
                i -= 1
            else:
                nums1[last_index] = nums2[j]
                j -= 1
            last_index -= 1
